ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/TimestampProtocolSpecAdapters.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public class TimestampProtocolSpecAdapters {
```

Comment:
Is this class even needed? It's implementation is the same as `ProtocolSpecAdapters`.

Reply:
Maybe not, think you're right. Probably could just rename ProtocolSpecAdapters' variable to work for both blockNumber and timestamp. Either way, this class will disappear when we play https://github.com/hyperledger/besu/issues/4788

Reply:
I've removed it and renamed ProtocolSpecAdapter variable to `blockNumberOrTimestamp` <-- I'm dubious about this though because it implies you can mix number and timestamp in the same modifiers Map. I'm not sure that could happen with the current implementation of TimestampSchedule. 
I'm also not sure if it's actually a problem to mix them either, would technically work as long as blocks don't follow timestamps.

An alternative approach would be to create a type hierarchy to discriminate between blockNumber and timestamp, but I've resisted this for now because the SpecAdapters won't be needed for mainnet/testnets with #4788
 
Another approach would be to leave ProtocolSpecAdapters as blockNumber only don't attempt to support other usecases such as Clique.

Reply:
Think it's ok to use `blockNumberOrTimestamp` for now.

Would prefer that it contained only one type as doesn't handle ordering between block numbers and timestamps unless timestamps strictly follow block numbers. Using generics and type hierarchy to enforce this would be nice so we could have `ProtocolSpecAdapters<Timestamp>` for example. But not sure it's worth the effort either as ProtocolSpecAdapters will go away soon for the timestamp case once #4788 is done.